packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1265,28 +1283,29 @@ class _TabBarViewState extends State<TabBarView> {
   @override
   void dispose() {
     if (_controllerIsValid)
-      _controller.animation.removeListener(_handleTabControllerAnimationTick);
+      _controller.removeAnimationStartedListener(_handleTabControllerAnimationStarted);
     _controller = null;
     // We don't own the _controller Animation, so it's not disposed here.
     super.dispose();
   }
 
   void _updateChildren() {
     _children = widget.children;
-    _childrenWithKey = KeyedSubtree.ensureUniqueKeysForList(widget.children);
+    _originalChildrenWithKey = KeyedSubtree.ensureUniqueKeysForList(widget.children);
+    _childrenWithKey = _originalChildrenWithKey;
   }
 
-  void _handleTabControllerAnimationTick() {
-    if (_warpUnderwayCount > 0 || !_controller.indexIsChanging)
+  void _handleTabControllerAnimationStarted(Duration duration, Curve curve) {
+    if (_warpUnderwayCount > 0 && _controller.index == _currentIndex)
       return; // This widget is driving the controller's animation.
 
```

Comment:
we may need to take care of where duration is null

Reply:
Yep, currently the duration and curve given are effectively NonNull because I passed in default values when not present. If we decide to pass in null instead, I'll take care of it here :)